CONTRIBUTING.MD

```diff
@@ -0,0 +1,101 @@
+pip install -e . -r dev-requirements.txt
+```
+
+When `dbt-snowflake` is installed this way, any changes you make to the `dbt-snowflake` source code will be reflected immediately in your next `dbt-snowflake` run.
+
+To confirm you have correct version of `dbt-core` installed please run `dbt --version` and `which dbt`.
+
+
+## Testing
+
+### Initial Setup
+
+`dbt-snowflake` contains [unit](https://github.com/dbt-labs/dbt-snowflake/tree/main/tests/unit) and [integration](https://github.com/dbt-labs/dbt-snowflake/tree/main/tests/integration) tests. Integration tests require tsting against an actual Snowflake warehouse. We have CI set up to test against a Snowflake warehouse. In order to run integration tests locally, you will need a `test.env` file in the root of the repository that contains credentials for Snowflake.
+
+Note: This `test.env` file is git-ignored, but please be _extra_ careful to never check in credentials or other sensitive information when developing. To create your `test.env` file, copy the provided example file, then supply your relevant credentials.
+
+```
+cp test.env.example test.env
+$EDITOR test.env
+```
+
+### Test commands
+There are a few methods for running tests locally.
+
+#### `tox`
+`tox` takes care of managing Python virtualenvs and installing dependencies in order to run tests. You can also run tests in parallel, for example you can run unit tests for Python 3.7, Python 3.8, Python 3.9, and `flake8` checks in parallel with `tox -p`. Also, you can run unit tests for specific python versions with `tox -e py37`. The configuration of these tests are located in `tox.ini`.
+
+#### `pytest`
+Finally, you can also run a specific test or group of tests using `pytest` directly. With a Python virtualenv active and dev dependencies installed you can do things like:
+
+```sh
+# run specific snowflake integration tests
+python -m pytest -m profile_snowflake tests/integration/simple_copy_test
+# run all unit tests in a file
+python -m pytest tests/unit/test_snowflake_adapter.py
+# run a specific unit test
+python -m pytest tests/unit/test_snowflake_adapter.py::TestSnowflakeAdapter::test_convert_date_type
+```
+## Updating Docs
+
+Many changes will require an update to the `dbt-snowflake` docs here are some useful resources.
+
+- Docs are [here](https://docs.getdbt.com/).
+- The docs repo for making changes is located [here]( https://github.com/dbt-labs/docs.getdbt.com).
+- The changes made are likely to impact one or both of [Snowflake Profile](https://docs.getdbt.com/reference/warehouse-profiles/snowflake-profile), or [Snowflake Configs](https://docs.getdbt.com/reference/resource-configs/snowflake-configs).
+- We ask every community member who makes a user-facing change to open an issue or PR regarding doc changes.
+
+
+
+## Submitting a Pull Request
+
+dbt Labs provides a CI environment to test changes to the `dbt-snowflake` adapter, and periodic checks against the development version of `dbt-core` through Github Actions.  
```

Comment:
what does this periodic checks part mean?

Reply:
I believe this is our nightly checks? not exactly sure if we have any triggers between the two otherwise maybe releases? times we would need to make sure changes to one doesn't break the other.

Reply:
I'm pretty sure if anything this would refer to checks whenever a commit is pushed as opposed to referring to our nightly monitoring checks. And hence, I'd suggest rewording this to be something like, "As mentioned above, dbt Labs....adapter. The GitHub Actions pipeline runs your feature branch using the current development version of `dbt-core`" -- something that indicates the CI pipeline uses the main branch in dbt core.